src/actinia_statistic_plugin/raster_sampling.py

```diff
@@ -0,0 +1,230 @@
+    """Perform raster map sampling on a raster map layer based on input points, asynchronous call
+    """
+    decorators = [log_api_call, auth.login_required]
+
+    def _execute(self, location_name, mapset_name, raster_name):
+
+        rdc = self.preprocess(has_json=True, has_xml=False,
+                              location_name=location_name,
+                              mapset_name=mapset_name,
+                              map_name=raster_name)
+        if rdc:
+            enqueue_job(self.job_timeout, start_job, rdc)
+
+        return rdc
+
+    @swagger.doc(deepcopy(SCHEMA_DOC))
+    def post(self, location_name, mapset_name, raster_name):
+        """Perform raster map sampling on a raster map layer based on input points asynchronously
+        """
+        self._execute(location_name, mapset_name, raster_name)
+        html_code, response_model = pickle.loads(self.response_data)
+        return make_response(jsonify(response_model), html_code)
+
+
+class SyncEphemeralRasterSamplingResource(AsyncEphemeralRasterSamplingResource):
+    """Perform raster map sampling on a raster map layer based on input points, synchronous call
+    """
+    decorators = [log_api_call, auth.login_required]
+
+    @swagger.doc(deepcopy(SCHEMA_DOC))
+    def post(self, location_name, mapset_name, raster_name):
+        """Perform raster map sampling on a raster map layer based on input points synchronously
+        """
+        check = self._execute(location_name, mapset_name, raster_name)
+        if check is not None:
+            http_code, response_model = self.wait_until_finish()
+        else:
+            http_code, response_model = pickle.loads(self.response_data)
+        return make_response(jsonify(response_model), http_code)
+
+
+def start_job(*args):
+    processing = AsyncEphemeralRasterSampling(*args)
+    processing.run()
+
+
+class AsyncEphemeralRasterSampling(EphemeralProcessing):
+    """Sample a raster map at vector points
+    """
+
+    def __init__(self, *args):
+        EphemeralProcessing.__init__(self, *args)
+        self.response_model_class = RasterSamplingResponseModel
+
+    def _execute(self):
+
+        self._setup()
+
+        # Points are stored in self.request_data
+        raster_name = self.map_name
+        points = self.request_data["points"]
+
+        if not points or len(points) == 0:
+            raise AsyncProcessError("Empty coordinate list")
+
+        point_file = tempfile.NamedTemporaryFile(dir=self.temp_file_path, delete=True)
+        result_file = tempfile.NamedTemporaryFile(dir=self.temp_file_path, delete=True)
+
+        for tuple in points:
+            if len(tuple) != 3:
+                raise AsyncProcessError("Wrong number of coordinate entries")
+
+            id, x, y = tuple
+            row = "%s|%s|%s\n" % (id, x, y)
+            point_file.write(row.encode())
+
+        point_file.flush()
+
+        pc = dict()
+        pc["1"] = {"module": "v.in.ascii",
```

Comment:
It would be nicer if the PC was written in the new format and not the old one, as we have an open issue in acitnia to no longer support the old format: https://github.com/mundialis/actinia_core/issues/245

Reply:
Sure, changed to new PC style in 214b495018bc9cfb928779892b710c1e0cd791e9